src/Context/product-context.js

```diff
@@ -0,0 +1,637 @@
+        genre: "philosophy",
+        rating : 3  
+    },
+    {
+        id : 17,
+        bookname : "I too had a love story",
+        author   : "Ravinder Singh",
+        originalprice : 400,
+        discountedprice : 300,
+        discountpercent : 25,
+        imgsrc : I_too_had_a_love_story,
+        imgalt : 'Book - I too had a love story',
+        badgetext : 'Romance',
+        outOfStock: false,
+        fastDeliveryAvailable : false,
+        genre: "romance",
+        rating : 3  
+    },
+    {
+        id : 18,
+        bookname : "The Mystery Of The Blue Train",
+        author   : "Agatha Christie",
+        originalprice : 400,
+        discountedprice : 300,
+        discountpercent : 25,
+        imgsrc : The_Mystery_Of_The_Blue_Train,
+        imgalt : 'Book - The Mystery Of The Blue Train',
+        badgetext : 'Thriller',
+        outOfStock: false,
+        fastDeliveryAvailable : false,
+        genre: "thriller",
+        rating : 4  
+    },
+    {
+        id : 19,
+        bookname : "Death On The Nile",
+        author   : "Agatha Christie",
+        originalprice : 400,
+        discountedprice : 300,
+        discountpercent : 25,
+        imgsrc : Death_On_The_Nile,
+        imgalt : 'Book - Death On The Nile',
+        badgetext : 'Thriller',
+        outOfStock: false,
+        fastDeliveryAvailable : false,
+        genre: "thriller",
+        rating : 4  
+    },
+    {
+        id : 20,
+        bookname : "Eloquent JavaScript",
+        author   : "Marijn Haverbeke",
+        originalprice : 800,
+        discountedprice : 600,
+        discountpercent : 25,
+        imgsrc : Eloquent_JavaScript,
+        imgalt : 'Book-Dan Brown - Eloquent JavaScript',
+        badgetext : 'Tech',
+        outOfStock: false,
+        fastDeliveryAvailable : true,
+        genre: "tech",
+        rating : 4   
+    },
+    {
+        id : 21,
+        bookname : "You Don't Know JS",
+        author   : "Kyle Simpson",
+        originalprice : 800,
+        discountedprice : 600,
+        discountpercent : 25,
+        imgsrc : You_Dont_Know_JS,
+        imgalt : 'Book-Dan Brown - You Dont Know JS',
+        badgetext : 'Tech',
+        outOfStock: false,
+        fastDeliveryAvailable : true,
+        genre: "tech",
+        rating : 4    
+    }
+]
+
+function productsOrderFunc(state,action)
+{
+    switch(action.type)
+    {
+        case "PRICE_HIGH_TO_LOW" : 
+            {
+                for(let i=0; i<state.length-1; i++)
+                {
+                    for(let j=0; j<state.length-1-i; j++)
+                    {
+                        if(state[j].discountedprice < state[j+1].discountedprice && i+1<state.length)
+                        {
+                            let swappedNum = state[j];
+                            state[j] = state[j+1];
+                            state[j+1] = swappedNum
+                        }
+                    }
+                }
```

Comment:
fix: You can use sort function here instead of doing this by bubble sort

Reply:
Yeah, great point, I am implementing Bubble sort and Chrome JS sort implements Quick and Insertion Sort, will update this. 😄

Reply:
Done.